tests/asynch/basic.rs

```diff
@@ -1,9 +1,103 @@
+use futures::try_join;
 use shuttle::sync::Mutex;
 use shuttle::{asynch, check_dfs, check_random, scheduler::PctScheduler, thread, Runner};
 use std::sync::atomic::{AtomicUsize, Ordering};
 use std::sync::Arc;
 use test_env_log::test;
 
+async fn do_err(e: bool) -> Result<(), ()> {
+    if e {
+        Err(())
+    } else {
+        Ok(())
+    }
+}
+
+// Check that try_join on shuttle futures works as expected
+#[test]
+fn test_try_join() {
+    check_dfs(
+        || {
+            let f2 = do_err(true);
+            let f1 = do_err(false);
+            let res = asynch::block_on(async { try_join!(f1, f2) });
+            assert!(res.is_err());
+        },
+        None,
+    );
+}
+
+// Check that when we block on a future, it gets executed in both executions
```

Comment:
I don't see how this test actually checks the *ordering* -- it just checks that the test ran twice and the blocked-on future ran twice.

Reply:
Do you think it would be better if the test showed that in exactly one of the two orderings, the blocked on task finishes first?

Reply:
No, I think just checking both executions is good enough, the comment was just a little misleading.